src/main/java/clients/SymBotClient.java

```diff
@@ -218,8 +225,26 @@ private void reportIfLoadBalanced(SymConfig config) {
     }
 
     public UserInfo getBotUserInfo() {
+        // botUserInfo is incomplete if getEmailAddress is null as we only parsed in username and id from JWT
+        if (botUserInfo == null || botUserInfo.getEmailAddress() == null) {
+            botUserInfo = this.getUsersClient().getSessionUser();
+        }
         return botUserInfo;
     }
+    
+    public String getBotUsername() {
+        if (botUserInfo == null || botUserInfo.getUsername() == null) {
+            getBotUserInfo();
+        }
+        return botUserInfo.getUsername();
+    }
+    
+    public long getBotUserId() {
+        if (botUserInfo == null || botUserInfo.getId() == null) {
+            getBotUserInfo();
+        }
+        return botUserInfo.getId();
+    }
```

Comment:
`SymBotClient` implements the `ISymClient` interface, are you sure that you don't want to move those new methods at interface level?

Reply:
These changes are specific to bots so should not be in ISymClient